generator/src/test/java/com/scottlogic/deg/generator/cucumber/interesting/InSet.feature

```diff
@@ -0,0 +1,144 @@
+    Given there is a field foo
+    And foo is in set:
+      | "ABC" |
+      | "DEF" |
+      | "GHI" |
+      | "JKL" |
+    Then the following data should be generated:
+      | foo   |
+      | null  |
+      | "ABC" |
+      | "DEF" |
+
+Scenario: Running an 'inSet' request that includes only a single string with alphabetic characters should return 1 value and null
+    Given there is a field foo
+    And foo is in set:
+      | "Test" |
+    Then the following data should be generated:
+      | foo    |
+      | null   |
+      | "Test" |
+
+Scenario: Running an 'inSet' request that includes 4 numeric values and not null should return 2 values
+    Given there is a field foo
+    And foo is anything but null
+    And foo is in set:
+      | 10 |
+      | 20 |
+      | 30 |
+      | 40 |
+    Then the following data should be generated:
+      | foo |
+      | 20  |
+      | 40  |
+
+Scenario: Running an 'inSet' request that includes 4 numeric values should return 2 values and null
+    Given there is a field foo
+    And foo is in set:
+      | 100 |
+      | 2000 |
+      | 300 |
+      | 550 |
+    Then the following data should be generated:
+      | foo  |
+      | null |
+      | 100  |
+      | 2000 |
+
+ Scenario: Running an 'inSet' request that includes 4 decimal numeric values should return 2 decimal values and null
+    Given there is a field foo
+    And foo is in set:
+      | 100.05   |
+      | 253.0054 |
+      | 10.8     |
+      | 0.00382  |
+    Then the following data should be generated:
+      | foo      |
+      | null     |
+      | 10.8     |
+      | 253.0054 |
+
+Scenario: Running an 'inSet' request that includes 4 temporal values should return 2 temporal values and null
+    Given there is a field foo
+    And foo is in set:
+      | "2010-01-01T00:00:00.000" |
+      | "2012-01-01T00:01:00.000" |
+      | "2011-01-02T00:00:00.000" |
+      | "2012-01-02T01:10:00.000" |
+    Then the following data should be generated:
+      | foo                       |
+      | null                      |
+      | "2010-01-01T00:00:00.000" |
+      | "2012-01-01T00:01:00.000" |
+
+Scenario: Running an 'inSet' request that has multiple special characters should return 2 values and null
+    Given there is a field foo
+    And foo is in set:
+      | "Cyrillic text: Тхис ис Тест Нумбер 01" |
+      | "Japanese text: これはテスト番号2です"   |
+      | "Korean text: 이것은 시험 번호 3입니다"  |
+    Then the following data should be generated:
+      | foo |
+      | null |
+      | "Japanese text: これはテスト番号2です"   |
+      | "Korean text: 이것은 시험 번호 3입니다"  |
+
+Scenario: Running an 'inSet' request that has multiple set constraints should return 2 values from each set and null
+    Given there is a field foo
+    And there is a constraint:
+      """
+      {
+        "anyOf": [
+          { "field": "foo", "is": "inSet", "values": [1, 2, 3, 4] },
+          { "field": "foo", "is": "inSet", "values": [5, 6, 7, 8] }
```

Comment:
I also noticed that we don't dedupe the results. Is this right? In the example below we will get 2 rows with null and two rows with 1. I don't think the duplicate rows add any value in terms of interesting. What do you think?

Scenario: Running an 'inSet' request that has multiple set constraints should return 2 values from each set (deduplicated) and null
    Given there is a field foo
    And there is a constraint:
      """
      {
        "anyOf": [
          { "field": "foo", "is": "inSet", "values": [1, 2, 3, 4] },
          { "field": "foo", "is": "inSet", "values": [1, 6, 7, 8] }
        ]
      }
      """
    Then the following data should be included in what is generated:
      | foo  |
      | null |
      | 1    |
      | 2    |
      | 6    |
      | 7    |

But this test fails with the below. Note the duplicate null and 1.

Expected: [null], [1], [2], [6], [7]
     but: [1], [2], [null], [1], [6], [null]
 missing: [7]

Reply:
That's correct because they will be treated as 2 separate specs. You get a duplicate for 1 because both sets has a 1 and a subset (2 items) is taken from each of the sets. For that same reason we get null as well

Reply:
From an end user perspective - are duplicate rows in interesting mode valuable? Would it be more interesting to have the same amount of rows but with different values?

Reply:
Hmm, interesting one.